scalyr_agent/builtin_monitors/url_monitor.py

```diff
@@ -103,11 +107,62 @@ def _initialize(self):
         else:
             self.extractor = None
 
-    def gather_sample(self):
+    @staticmethod
+    def form_request(url, request_method, request_data, headers, logger):
+        """
+        Forms the HTTP request based on the request URL, HTTP headers and method
+        @param url: HTTP request URL
+        @param request_method: HTTP request method
+        @param request_data: HTTP request payload data
+        @param headers: HTTP header
+        @param logger: The logger to use
+
+        @type url: str
+        @type request_method: str
+        @type request_data: str | None
+        @type headers: JsonArray (of tuples with header name, header value combination)
+        @type logger: logging.Logger or subtypes
+        @return: Request object
+        """
+
+        request = urllib2.Request(url, data=request_data)
+        if headers:
+            if type(headers) != JsonArray:
```

Comment:
Let's move this check for `type(headers) != JsonArray` to the constructor.  If there is a configuration error, we try to throw as soon as we can (like in the constructor) because the rest of the system behaves better if it does.  (The system is designed to expect configuration errors to be thrown during monitor construction).